src/dmd/dcast.d

```diff
@@ -1467,27 +1467,6 @@ Type toStaticArrayType(SliceExp e)
     return null;
 }
 
-// Try casting the alias this member. Return the expression if it succeeds, null otherwise.
-private Expression tryAliasThisCast(Expression e, Scope* sc, Type tob, Type t1b, Type t)
-{
-    Expression result;
-    AggregateDeclaration t1ad = isAggregate(t1b);
-    if (!t1ad)
-        return null;
-
-    AggregateDeclaration toad = isAggregate(tob);
-    if (t1ad == toad || !t1ad.aliasthis)
-        return null;
```

Comment:
What happened to lines 1473-1480?

Reply:
Look at the URL in the first comment, they were redundant and had 0 coverage according to test suite.

Reply:
Maybe that could be done in the PR that is fixing the issue.